shim.c

```diff
@@ -1640,8 +1640,12 @@ static EFI_STATUS generate_path_from_image_path(EFI_LOADED_IMAGE *li,
 		bootpath[j] = '\0';
 	}
 
-	while (*ImagePath == '\\')
-		ImagePath++;
+	for (i = 0, last = -1; i < StrLen(ImagePath); i++)
+		if (ImagePath[i] == '\\')
+			last = i + 1;
+
+	if (last >= 0)
```

Comment:
@maranwilson thanks for the test and review.

right, I had `>= 0` because initially assigned `last = i` and `ImagePath = ImagePath + last + 1`. Later I changed the assignments but forgot to update the condition.